syntax = "proto3";

package renderserver;

service RenderServer {
    // Used to signal that a new animation is ready to be animated.
    rpc AnimationStatus (AnimationStatusRequest) returns (AnimationStatusResponse);

    // Used to signal when manim starts and stops.
    rpc ManimStatus (ManimStatusRequest) returns (ManimStatusResponse);
}

// Signifies that the next animation is ready to be rendered.
message AnimationStatusRequest {}
message AnimationStatusResponse {}

message ManimStatusRequest {
    // The name of the scene whose frames are currently being served. This is
    // only seen if a javascript renderer is running prior to manim being
    // started.
    string scene_name = 1;

    // Indicates that the current Scene is finished.
    bool scene_finished = 2;
}
message ManimStatusResponse {}
